src/utils/PersonIdentityUtils.ts

```diff
@@ -21,22 +20,28 @@ export const personIdentityUtils = {
 		const { givenNames, familyNames } = this.getNames(personDetails);
 		const f2fGivenNames = givenNames.join(" ");
 		const f2fFamilyName = familyNames.join(" ");
-
+    
 		// Check if the fullName in f2f matches the fullName in DocumentFields
 		const doesFullNameMatch = `${f2fGivenNames.toLowerCase()} ${f2fFamilyName.toLowerCase()}` === yotiFullName.toLowerCase();
 		if (!doesFullNameMatch) {
 			const errorMessage = "FullName mismatch between F2F & YOTI";
 			logger.error({ message: errorMessage }, { messageCode: MessageCodes.VENDOR_SESSION_NAME_MISMATCH });
 			throw new AppError(HttpCodesEnum.SERVER_ERROR, errorMessage);
 		}
-
-		// Remove family name from the full name and split at spaces
-		const yotiGivenNameParts = yotiFullName.replace(new RegExp(f2fFamilyName, "i"), "").match(/\S+/g);
-		// Map the array of given names into the correct format
-		const nameParts = yotiGivenNameParts.map((name: string) => ({ value: name.trim(), type: "GivenName" }));
-		// Remove the given names from the full name, remove surrounding spaces, and map to correct format
-		nameParts.push({ value: yotiFullName.replace(new RegExp(f2fGivenNames, "i"), "").trim(), type: "FamilyName" });
-
+    
+		// Split the full name into words
+		const fullNameWords = yotiFullName.split(" ").filter((part: string) => part !== "");
+    
+		// Split the words into given names and family name parts
+		const nameParts = [];
+		for (const word of fullNameWords) {
+			if (f2fFamilyName.toLowerCase().includes(word.toLowerCase())) {
+				nameParts.push({ value: word.trim(), type: "FamilyName" });
+			} else {
+				nameParts.push({ value: word.trim(), type: "GivenName" });
+			}
```

Comment:
Does this mean that for any parts of the yoti full name that aren't in the family name that we have, we are assuming they are given names? What if it's just a genuine mismatch? 

Eg yoti full name is `boaty mcboatface boat-face` but f2f name is `boaty mcboatface boat`, will this code mean that `boat-face` is pushed as a given name?

Reply:
Eg yoti full name is boaty mcboatface boat-face but f2f name is boaty mcboatface boat 
This will fail on line 25 and throw an error
`const doesFullNameMatch = `${f2fGivenNames.toLowerCase()} ${f2fFamilyName.toLowerCase()}` === yotiFullName.toLowerCase();
		if (!doesFullNameMatch) {
			const errorMessage = "FullName mismatch between F2F & YOTI";
			logger.error({ message: errorMessage }, { messageCode: MessageCodes.VENDOR_SESSION_NAME_MISMATCH });
			throw new AppError(HttpCodesEnum.SERVER_ERROR, errorMessage);
		}`

Reply:
Ah nice thanks!